Source/RunActivity/Program.cs

```diff
@@ -65,7 +65,7 @@ static void Main(string[] args)
             game.PushState(new GameStateRunActivity(args));
             game.Run();
 
-            settings.Save();
+            settings.SaveInGameChanges();
```

Comment:
An objection to this whole approach is that saving on exit requires the program to exit cleanly, including the computer not crashing.

And targeting these specific settings is contrary to what the existing code does, which is explicitly saving when things are changed: https://github.com/openrails/openrails/blob/69600d7daa88e3340bebe894aaa0f226b81d3495/Source/RunActivity/Viewer3D/Popups/Window.cs#L105 (granted that also doesn't respect `/skip-user-settings` but that would be an easy fix too)

While this PR improves the behaviour of the newly added saving of settings, it is still disjoint from the pre-existing method and I would like to see that addressed somewhere.